### DISCUSSION TITLE
请教 reactive-react 中的 useDidUpdate 的作用

### DISCUSSION THREAD

**Discussion by UlricGan** [alibaba/formily#1481] [Q&A 2.x]
具体有什么作用？为什么这么实现？immediate 可以用 setTimeout 替代吗？

背景：字节小程序的iOS端发现 useDidUpdate 中的 callback 执行了两次（导致forceUpdate不执行），替换成 setTimeout 后没问题了；安卓端没有上述问题

**Comment by janryWang** [comment]
其实就是一个比setTimeout更快的异步函数，promise是走了microtask
